oeps/oep-0034-bp-lint-amnesty.rst

```diff
@@ -0,0 +1,78 @@
+====================
+OEP-34: Lint Amnesty
+====================
+
++-----------------+--------------------------------------------------------+
+| OEP             | :doc:`OEP-34 </oeps/oep-0034-bp-lint-amnesty>`         |
++-----------------+--------------------------------------------------------+
+| Title           | Lint Amnesty                                           |
++-----------------+--------------------------------------------------------+
+| Last Modified   | 2019-03-12                                             |
++-----------------+--------------------------------------------------------+
+| Authors         | Calen Pennington <cale@edx.org>                        |
++-----------------+--------------------------------------------------------+
+| Arbiter         |  TBD                                                   |
++-----------------+--------------------------------------------------------+
+| Status          | Draft                                                  |
++-----------------+--------------------------------------------------------+
+| Type            | Best Practice                                          |
++-----------------+--------------------------------------------------------+
+| Created         | 2019-03-12                                             |
++-----------------+--------------------------------------------------------+
+| `Review Period` | 2019-03-12 - 2019-04-05                                |
++-----------------+--------------------------------------------------------+
+
+Context
+-------
+
+All software has bugs. Linting can identify a subset of those bugs (and
+stylistic issues) statically before the code is merged to master or
+released to production. At edX, we strive to keep violations minimal or
+at zero, but for certain codebases, such as edx-platform, we have a large
+backlog of violations. We use a cap to limit adding new violations, but
+it must be updated manually, and so it only incrementally forces the
+number of violations down. We also disallow new code changes that introduce
+violations *in the lines that were changed*
+
+Certain violations are dependent on a context larger than just the line
+that the warning is on. For instance, a code change might delete a class
+that is imported in a number of other code files. Those imports would now
+be in violation, but they were not changed, and therefor would not be
+flagged by differential linting.
+
+Decision
+--------
+
+In all repositories, we will grant amnesty to all existing linting violations.
+This includes pylint, xsslint, pycodestyle, and any other linting programs
```

Comment:
Generally love the idea, but I have some hesitation around xsslint, especially when it comes to repos that haven't yet employed it.

Reply:
I guess my feeling is that adding the ignores is no worse (and probably better) than having cap on the number. Either way, the issues aren't being immediately fixed, and I think flagging them in the code makes them more visible.

Reply:
The xsslint issues are being fixed. I'd still recommend not listing this, but I'll leave that to the security team to determine.

Reply:
@feanil: It looks like this got merged, but I still propose that xsslint gets dropped from this list.